test/ec_comp_test.c

```diff
@@ -2066,235 +86,369 @@ static int init_curves(void)
+    EC_GROUP_free(g);
+    return NULL;
 }
 
-static EC_KEY *get_existing_key(size_t curve_i, size_t param_format_i, size_t comp_format_i, struct default_config_st *dcfg)
+static EC_KEY *get_existing_key(size_t curve_i, size_t param_format_i,
+                                size_t comp_format_i, size_t obj_type)
 {
-    EC_KEY *k = NULL;
+    EC_KEY *k = NULL, *res = NULL;
+    EC_GROUP *g = NULL;
     FILE *f = NULL;
     BIO *b = NULL;
     char filename[256];
-    snprintf(filename, sizeof(filename), "test/recipes/15-test_ec_comp_data/%s-%s-%s.priv",
-        known_curve_names[curve_i],
-        param_formats_i[param_format_i] < 0 ? "named_curve" : param_formats[param_format_i],
-        comp_formats_i[comp_format_i] < 0 ? "uncompressed" : comp_formats[comp_format_i]);
+
+    OPENSSL_assert(param_formats_i[param_format_i] >= 0);
+    OPENSSL_assert(comp_formats_i[comp_format_i] >= 0);
+
+    if (obj_type == OBJ_TYPE_PARAMS)
+        snprintf(filename, sizeof(filename), "test/recipes/15-test_ec_comp_data/%s-%s.param",
+                 known_curve_names[curve_i],
+                 param_formats[param_format_i]);
+    else
+        snprintf(filename, sizeof(filename), "test/recipes/15-test_ec_comp_data/%s-%s-%s.priv",
+                 known_curve_names[curve_i],
+                 param_formats[param_format_i],
+                 comp_formats[comp_format_i]);
 
     b = BIO_new(BIO_s_file());
     if (b == NULL)
         goto fail;
 
     f = fopen(filename, "rb");
-    if (f == NULL) {
-        fprintf(stderr, "# Could not open '%s'\n", filename);
+    if (f == NULL)
         return NULL;
-    }
 
     BIO_set_fp(b, f, BIO_NOCLOSE);
 
-    if (PEM_read_bio_ECPrivateKey(b, &k, NULL, NULL) == 0)
-        goto fail;
+    if (obj_type == OBJ_TYPE_PARAMS) {
+        if (PEM_read_bio_ECPKParameters(b, &g, NULL, NULL) == 0)
+            goto fail;
 
-    dcfg->asn1_flag  = EC_GROUP_get_asn1_flag(EC_KEY_get0_group(k));
-    dcfg->point_form = EC_GROUP_get_point_conversion_form(EC_KEY_get0_group(k));
+        k = EC_KEY_new();
+        if (k == NULL)
+            goto fail;
 
-    if (param_formats_i[param_format_i] >= 0)
-        EC_KEY_set_asn1_flag(k, param_formats_i[param_format_i]);
-    if (comp_formats_i[comp_format_i] >= 0)
-        EC_KEY_set_conv_form(k, comp_formats_i[comp_format_i]);
+        if (EC_KEY_set_group(k, g) == 0)
+            goto fail;
 
-    BIO_free(b);
-    fclose(f);
-    return k;
+        EC_GROUP_free(g);
+        g = NULL;
+    } else {
+        if (PEM_read_bio_ECPrivateKey(b, &k, NULL, NULL) == 0)
+            goto fail;
+    }
 
+    res = k;
+    k = NULL;
 fail:
+    EC_GROUP_free(g);
     EC_KEY_free(k);
     BIO_free(b);
     fclose(f);
-    return NULL;
-    //return generate_new_key(curve_i, asn1_flag, point_form);
+    return res;
 }
 
-static int should_skip(size_t curve_i, size_t param_format_i)
+static int verify_expected(int param_format, int comp_format, int asn1_flag, int point_form, int obj_type)
```

Comment:
Maybe add a comment here to indicate which input are "expected values" and which are the "values under examination".